env.sh

```diff
@@ -1,71 +1,79 @@
 #!/usr/bin/env bash
 
-DOCKER_IMAGE="cita/cita-build:ubuntu-18.04-20190304"
-if [[ `uname` == 'Darwin' ]]
-then
-    cp /etc/localtime $PWD/localtime
-    LOCALTIME_PATH="$PWD/localtime"
+SOURCE_DIR="$(cd $(dirname "$0") && pwd -P)"
+
+test -f "${SOURCE_DIR}/CODE_OF_CONDUCT.md"
+if [ $? -eq 0 ]; then
+    CONTAINER_NAME="cita_build_container"
+    DOCKER_IMAGE="cita/cita-build:ubuntu-18.04-20190304"
 else
-    LOCALTIME_PATH="/etc/localtime"
+    CONTAINER_NAME="cita_run_container"
+    DOCKER_IMAGE="cita/cita-run:ubuntu-18.04-20181009"
+    SOURCE_DIR="$(dirname $SOURCE_DIR)"
 fi
 
-docker_bin=$(which docker)
-if [ -z "${docker_bin}" ]; then
-    echo "Command not found, install docker first."
-    exit 1
+if [ `uname` == 'Darwin' ]; then
+    cp /etc/localtime ${SOURCE_DIR}/localtime
+    SYSTEM_NET="bridge"
+    LOCALTIME_PATH="${SOURCE_DIR}/localtime"
 else
-    docker version > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        echo "Run docker version failed, Maybe docker service not running or current user not in docker user group."
-        exit 2
-    fi
+    SYSTEM_NET="host"
+    LOCALTIME_PATH="/etc/localtime"
 fi
 
-SOURCE_DIR=`pwd`
-CONTAINER_NAME="cita_build${SOURCE_DIR//\//_}"
-CARGO_HOME=/opt/.cargo
 WORKDIR=/opt/cita
 USER_ID=`id -u $USER`
 USER_NAME="user"
 
-if [ "${USER_ID}" = "0" ]; then
-    USER_NAME="root"
-fi
+[[ "${USER_ID}" = "0" ]] && USER_NAME="root"
 
-mkdir -p ${HOME}/.docker_cargo/git
-mkdir -p ${HOME}/.docker_cargo/registry
+# Init contanier's cargo, for logs.
+CARGO_HOME=/opt/.cargo
+DOCKER_CARGO=${HOME}/.docker_cargo
+mkdir -p ${DOCKER_CARGO}/git
+
+# Port condition
+EXPOSE="1337:1337"
+if [ "$3" == "port" ]; then
+    EXPOSE=${@:4}
+    [[ "${EXPOSE}" = "NULL" ]] && EXPOSE="1337:1337"
+    docker container stop $CONTAINER_NAME > /dev/null 2>&1
+fi
 
+# Run container
 docker ps | grep ${CONTAINER_NAME} > /dev/null 2>&1
 if [ $? -ne 0 ]; then
     echo "Start docker container ${CONTAINER_NAME} ..."
     docker rm ${CONTAINER_NAME} > /dev/null 2>&1
 
-    # test network and set init cmd
-    timeout=3
-    target=www.google.com
-    ret_code=`curl -I -s --connect-timeout $timeout $target -w %{http_code} | tail -n1`
-    if [ "x$ret_code" = "x200" ]; then
-        INIT_CMD="while true;do sleep 100;done"
-    else
-        INIT_CMD="echo -e '[source.crates-io]\nregistry = \"https://github.com/rust-lang/crates.io-index\"\nreplace-with = \"ustc\"\n[source.ustc]\nregistry = \"https://mirrors.ustc.edu.cn/crates.io-index\"' | sudo tee /opt/.cargo/config;while true;do sleep 100;done"
-    fi
-
```

Comment:
I not sure about these lines, I built failed when I use `ustc` sources, the same issue can refer to [#6513][1] in cargo.

[1]: https://github.com/rust-lang/cargo/issues/6513

Reply:
Would you please to create another issue about this.
I think it's better to fix it on another pr.